### PR TITLE
3D Scene consuming 100% CPU resources issue fix (2)

### PR DESCRIPTION
**Pull request by enicolasgomez**
had to create a new PR than #150 as started fresh from a new branch, here's the change log:

1. The animation frame on ThreeDEngine is not a recursive loop anymore, this is the starting point to solve this issue
2. TrackballControls has been kept in favor of OrbitControls
3. ThreeDEngine instances is now exposed to TrackballControl component
4. Internally, Trackballcontrol uses ThreeDEngine.requestAnimaionFrame() internally on `update()` only when the lastposition of the camera has changed
5. `defaultCameraControlsHandler` on Canvas will end up requesting an animation frame. This is needed as there's no a constant loop anymore
6. handlers have been added to Canvas React component to requestAnimationFrame() upon doucment loading

#150 
https://github.com/MetaCell/open-physiology-viewer/issues/31

### CODE REVIEW

**Comment by enicolasgomez** [comment]
One thing that worth mentioning as an improvement is subscribing to TrackballControls events from ThreeDEngine and fire the update event from there, internally to keep trackballcontrols agnostic from the engine.

**Review by afonsobspinto — CHANGES_REQUESTED** [review]
Seems like the previous issues were fixed, thanks for that :+1: 

However, and I'm not sure if this was already happening in the previous PR, now I noticed that the selection action is one frame behind / delayed (perhaps not triggering the requestFrame method?). Check below:

demo

Only after changing the camera, the selected object changes color.

Besides that, I agree with your suggestion of using events to loose the coupling between the engine and the trackball controls. I would see this change as a nice to have and not something mandatory to approve.

**Comment by enicolasgomez** [comment]
- Using events to loose the coupling between the engine and the trackball
- Plugged in intersect event to rendering 
- Incremented autorotate timeout to increment FPS

File: geppetto.js/geppetto-ui/src/3d-canvas/threeDEngine/ThreeDEngine.js [hunk 1]
```diff
@@ -148,6 +148,7 @@ export default class ThreeDEngine {
       this.cameraManager.getCamera(),
       this.renderer.domElement,
       this.cameraHandler,
+      this
```
Comment (afonsobspinto) [inline geppetto.js/geppetto-ui/src/3d-canvas/threeDEngine/ThreeDEngine.js:151]: No longer needed

**Pull request merged**